gapil/analysis/reference_value.go

```diff
@@ -184,7 +186,9 @@ func (v *ReferenceValue) field(s *scope, name string) Value {
 // in the scope's instances map.
 func (v *ReferenceValue) setField(s *scope, name string, val Value) Value {
 	for a := range v.Assignments {
-		s.instances[a] = s.getInstance(a).(fieldHolder).setField(s, name, val)
+		if fh, ok := s.getInstance(a).(fieldHolder); ok {
```

Comment:
Same problem as above

Reply:
After much beard scratching, I think I know what the issue is. Your fix is likely fine, however please add the below comment so that some poor bugger doesn't have to work this out again.

I suggest adding this new utility function, and using it in the two places you've changed:

``` go

func (v *ReferenceValue) fhAssignments() []fieldHolder {
	out := make([]fieldHolder, 0, len(v.Assignments))
	for a := range v.Assignments {
		// Check the assignment has an instance. No instance can happen when the 
		// assignment took place in a sibling block which has not been merged 
		// into the common scope yet.
		// In this particular case, the instance is inaccessible to this block, 
		// so skipping it is the correct thing to do.
		if fh, ok := s.getInstance(a).(fieldHolder); ok {
			out = append(out, fh)
		}
	}
	return out
}
```